test/components/legendTests.ts

```diff
@@ -350,3 +350,183 @@ describe("Legend", () => {
     svg.remove();
   });
 });
+
+describe("InterpolatedColorLegend", () => {
+  var svg: D3.Selection;
+  var colorScale: Plottable.Scale.InterpolatedColor;
+
+  beforeEach(() => {
+    svg = generateSVG(400, 400);
+    colorScale = new Plottable.Scale.InterpolatedColor();
+  });
+
+  function assertBasicRendering(legend: Plottable.Component.InterpolatedColorLegend) {
+    var scaleDomain = colorScale.domain();
+    var legendElement: D3.Selection = (<any> legend)._element;
+
+    var swatches = legendElement.selectAll(".swatch");
+    assert.strictEqual(d3.select(swatches[0][0]).attr("fill"),
+                       colorScale.scale(scaleDomain[0]),
+                       "first swatch's color corresponds with first domain value");
+    assert.strictEqual(d3.select(swatches[0][swatches[0].length - 1]).attr("fill"),
+                       colorScale.scale(scaleDomain[1]),
+                       "last swatch's color corresponds with second domain value");
+
+    var swatchContainer = legendElement.select(".swatch-container");
+    var swatchContainerBCR = swatchContainer.node().getBoundingClientRect();
+    var swatchBoundingBox = legendElement.select(".swatch-bounding-box");
+    var boundingBoxBCR = swatchBoundingBox.node().getBoundingClientRect();
+    assert.isTrue(Plottable._Util.DOM.boxIsInside(swatchContainerBCR, boundingBoxBCR),
+                  "bounding box contains all swatches");
+
+    var elementBCR = legendElement.node().getBoundingClientRect();
+    assert.isTrue(Plottable._Util.DOM.boxIsInside(swatchContainerBCR, elementBCR),
+                  "swatches are drawn within the legend's element");
+
+    var formattedDomainValues = scaleDomain.map((<any> legend)._formatter);
+    var labels = legendElement.selectAll("text");
+    var labelTexts = labels[0].map((textNode: HTMLScriptElement) => textNode.textContent);
+    assert.deepEqual(labelTexts, formattedDomainValues, "formatter is used to format label text");
+  }
```

Comment:
The problem with the current structure is that the tests that are higher up will get priority which is ok if the tests are similar since priority has a less of an effect, but if the tests are not similar then you'll only have information from the first run error.